pkg/surveyjson/survey.go

```diff
@@ -352,8 +352,29 @@ func (o *JSONSchemaOptions) recurse(name string, prefixes []string, requiredFiel
 			return err
 		}
 
-		// TODO ref path would need to be traversed to get sub objects, assuming right now that all defs are one level deep
-		err = o.recurse(name, make([]string, 0), make([]string, 0), definitions[refPath[0]], nil, output, make([]survey.Validator, 0), existingValues, definitions)
+		currentObject := (*definitions[refPath[0]]).(map[string]interface{})
+		for i := 1; i < len(refPath); i += 1 {
+			object, ok := currentObject[refPath[i]]
+			if !ok {
+				return errors.New(fmt.Sprintf("Could not resolve ref path \"%v\" is not a key in the object", refPath[i]))
+			}
+
+			currentObject = object.(map[string]interface{})
```

Comment:
Note that we should have extra variable `ok` to check if cast even works. 
Without that we get nil pointer reference.